src/Sarif.Converters/FxCopConverter.cs

```diff
@@ -108,7 +137,7 @@ internal Result CreateResult(FxCopLogReader.Context context)
             }
 
             result.RuleId = context.CheckId;
-            result.Message = !string.IsNullOrWhiteSpace(context.Message) ? new Message { Text = context.Message } : null;
+            result.Message = new Message { Arguments = context.Items, MessageId = context.ResolutionName, Text = context.Message };
```

Comment:
>MessageId = context.ResolutionName, Text = context.Message [](start = 70, length = 58)

The message text can be derived from the message format string + arguments. Do you want to include both? This was the same question I asked when I reviewed the SARIF you gen'd up for <redacted>.

Reply:
I see. The code below says that a project file produces a fully formed message, whereas a report file produces a message format string + arguments. So you'll never have both. That is totally worth a comment here.

---
In reply to: [212451123](https://github.com/Microsoft/sarif-sdk/pull/1044#discussion_r212451123) [](ancestors = 212451123)